packages/babel-parser/src/parser/statement.js

```diff
@@ -2042,6 +2058,27 @@ export default class StatementParser extends ExpressionParser {
     node.specifiers.push(this.finishNode(specifier, type));
   }
 
+  maybeParseModuleAttributes() {
+    if (!this.eat(tt._with)) {
+      if (this.hasPlugin("moduleAttributes")) return [];
+      return null;
+    }
+    this.expectPlugin("moduleAttributes");
+    const attrs = [];
+    do {
+      const node = this.startNode();
+      node.key = this.parseIdentifier();
+      node.computed = false;
+      node.method = false;
+      node.shorthand = false;
+      this.expect(tt.colon);
+      node.value = this.parsePrimitiveValue();
+      this.finishNode(node, "ObjectProperty");
```

Comment:
I don't think that we should re-use `ObjectProperty`: it's conceptually and syntactically different. We should introduce a new `ModuleAttribute` node.

Reply:
Would it help if we agreed on the Babel AST before the implemation?

Reply:
Changed the code to now use the new AST node `ImportAttribute`